src/SpeedyTransforms/spectral_transform.jl

```diff
@@ -196,65 +196,66 @@ function SpectralTransform(
     solid_angles = get_solid_angles(Grid, nlat_half)
 
     # RECURSION FACTORS
-    ϵlms = get_recursion_factors(lmax+1, mmax)
+    ϵlms = get_recursion_factors(spectrum)
 
     # GRADIENTS (on unit sphere, hence 1/radius-scaling is omitted)
     # meridional gradient for scalars (coslat scaling included)
-    grad_y1 = zeros(LowerTriangularMatrix, lmax+1, mmax+1)      # term 1, mul with harmonic l-1, m
-    grad_y2 = zeros(LowerTriangularMatrix, lmax+1, mmax+1)      # term 2, mul with harmonic l+1, m
+    grad_y1 = zeros(LowerTriangularMatrix, spectrum)      # term 1, mul with harmonic l-1, m
+    grad_y2 = zeros(LowerTriangularMatrix, spectrum)      # term 2, mul with harmonic l+1, m
```

Comment:
For RingGrids the equivalent is now just `zeros(grid)`, given that `grid` implies the actual type. Similar I could think one could just write `zeros(spectrum)` here, or `zeros(NF, spectrum)`?

Reply:
Yes, those constructors already exist. I just didn't use them everywhere in the code yet. 
* `zeros(LowerTriangularMatrix, spectrum)`
* `zeros(spectrum)`
*  `zeros(LowerTriangularMatrix, 5,5)`

Is all possible